WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewViewModel.swift

```diff
@@ -301,8 +317,25 @@ private extension ProductDetailPreviewViewModel {
                                                     existingCategories: existingCategories,
                                                     existingTags: existingTags)
 
-        let categories = existingCategories.filter({ aiProduct.categories.contains($0.name) })
-        let tags = existingTags.filter({ aiProduct.tags.contains($0.name) })
+        var categories = [ProductCategory]()
+
+        aiProduct.categories.forEach { aiCategory in
+            if let match = existingCategories.first(where: { $0.name == aiCategory }) {
+                categories.append(match)
+            } else {
+                categories.append(ProductCategory(categoryID: 0, siteID: siteID, parentID: 0, name: aiCategory, slug: ""))
+            }
+        }
+
+        var tags = [ProductTag]()
+
+        aiProduct.tags.forEach { aiTag in
+            if let match = existingTags.first(where: { $0.name == aiTag }) {
+                tags.append(match)
+            } else {
+                tags.append(ProductTag(siteID: siteID, tagID: 0, name: aiTag, slug: ""))
+            }
+        }
```

Comment:
Nit: This new logic looks complicated, could you add a comment on the containing method explaining how we are processing the suggested categories and tags? That would make it easier to maintain the code in the future.

Reply:
I agree. Added comments in https://github.com/woocommerce/woocommerce-ios/pull/10864/commits/2819161fb6d26361cc3416b1b7752c7901c796a0